opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -155,32 +297,66 @@ class ParallelOverlappingILU0
     virtual void apply (Domain& v, const Range& d)
     {
         Range& md = const_cast<Range&>(d);
-        comm_.copyOwnerToAll(md,md);
-        auto endrow=ilu_.end();
-        for ( auto row = ilu_.begin(); row != endrow; ++row )
+        copyOwnerToAll( md );
+
+        // iterator types
+        typedef typename Range ::block_type  dblock;
+        typedef typename Domain::block_type  vblock;
+
+        const size_type iEnd = lower_.rows();
+        const size_type lastRow = iEnd - 1;
+        if( iEnd != upper_.rows() )
         {
-            auto rhs(d[row.index()]);
-            for ( auto col = row->begin(); col.index() < row.index(); ++col )
-            {
-                col->mmv(v[col.index()],rhs);
-            }
-            v[row.index()] = rhs;
+            std::abort();
```

Comment:
Do we really want a `std::abort()` here? Looks like it is a bug in the implementation should the condition ever hold. `assert(iEnd==upper_.rows())` to the rescue?